File: insight-brain-service/src/main/frontend/dashboard/dashboard.module.js
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
/* global angular */

import dashboardUtilsModule from './utils/dashboard.utils.module';
import dashboardDataService from './services/dashboard.data.service';
import policyTrendController from './results/PolicyTrendController';
import violationsTableRow from './results/violationsTableRow';
import dashboardResultsController from './results/dashboard.results.controller';
import dashboardFilterModule from './filter/module';

var dashboardModule = angular.module('dashboard.module',
    [
      'ui.router', 'Stores', 'AngularCommon', 'ComponentModule', 'ComponentDisplay', dashboardUtilsModule.name,
      'utility', dashboardFilterModule.name
    ])
    .service('dashboard.data.service', dashboardDataService)

    // dashboard results
    .controller('PolicyTrendController', policyTrendController)
    .component('violationsTableRow', violationsTableRow)
    .controller('dashboard.results.controller', dashboardResultsController);

export default dashboardModule;

// To avoid hacking dependency order, states must be declared with their parent.
// Fixed https://github.com/angular-ui/ui-router/pull/492
dashboardModule.config(['$stateProvider', '$urlRouterProvider', function($stateProvider, $urlRouterProvider) {

  $stateProvider.state('dashboard', {
    url: '/dashboard',
    abstract: true,
    templateUrl: 'dashboard/dashboard.view.html?' + clmBuildTimestamp,
    data: {
      title: 'Dashboard',
      crumb: 'Dashboard'
    }
  }).state('dashboard.overview', {
    url: '?timeFilterFeature', // query parameter feature flag for Time-based filter control
    abstract: true,
    views: {
      content: {
        templateUrl: 'dashboard/results/dashboard.results.html?' + clmBuildTimestamp,
        controller: 'dashboard.results.controller'
      },
      filter: 'dashboardFilter'
    }
  }).state('dashboard.overview.violations', {
    url: '/violations',
    views: {
      'dashboard-results': {
        templateUrl: 'dashboard/results/violations.html?' + clmBuildTimestamp
      }
    },
    data: {
      title: 'Violations'
    }
  }).state('dashboard.overview.components', {
    url: '/components',
    views: {
      'dashboard-results': {
        templateUrl: 'dashboard/results/components.html?' + clmBuildTimestamp
      }
    },
    data: {
      title: 'Components'
    }
  }).state('dashboard.overview.applications', {
    url: '/applications',
    views: {
      'dashboard-results': {
        templateUrl: 'dashboard/results/applications.html?' + clmBuildTimestamp
      }
    },
    data: {
      title: 'Applications'
    }
  }).state('dashboard.component', {
    url: '/component/{hash}',
    controller: 'componentController',
    templateUrl: 'dashboard/component.html?' + clmBuildTimestamp,
    data: {
      crumb: 'Component Details'
    }
  });

  $urlRouterProvider.when('/dashboard/newest-risk', '/dashboard/violations');
}]);

File: insight-brain-service/src/main/java/com/sonatype/insight/brain/service/modules/DashboardModule.java
/*
 * Copyright (c) 2011-present Sonatype, Inc. All rights reserved.
 * Includes the third-party code listed at http://links.sonatype.com/products/clm/attributions.
 * "Sonatype" is a trademark of Sonatype, Inc.
 */
package com.sonatype.insight.brain.service.modules;

import com.sonatype.insight.brain.dashboard.DashboardFilterService;
import com.sonatype.insight.brain.dashboard.DashboardUtils;
import com.sonatype.insight.brain.dashboard.H2ApplicationRiskService;
import com.sonatype.insight.brain.dashboard.H2ComponentRiskService;
import com.sonatype.insight.brain.dashboard.H2DashboardViolationRiskService;
import com.sonatype.insight.brain.dashboard.PostgresApplicationRiskService;
import com.sonatype.insight.brain.dashboard.PostgresComponentRiskService;
import com.sonatype.insight.brain.dashboard.PostgresDashboardViolationRiskService;

import com.google.inject.AbstractModule;

/**
 * Guice module providing explicit bindings for Dashboard components. This replaces Sisu's automatic @Named component
 * discovery.
 */
public class DashboardModule
    extends AbstractModule
{
  @Override
  protected void configure() {
    bind(DashboardFilterService.class);
    bind(DashboardUtils.class);
    bind(H2ApplicationRiskService.class);
    bind(H2ComponentRiskService.class);
    bind(H2DashboardViolationRiskService.class);
    bind(PostgresApplicationRiskService.class);
    bind(PostgresComponentRiskService.class);
    bind(PostgresDashboardViolationRiskService.class);
  }
}
